### PR TITLE
Add new entry point to PBRCustomMaterial and a isEnabled property to Layer

### PR DESCRIPTION
**Pull request by Popov72**
Need these for the WebGPU demo!

I have also added the possibility to read the data from a texture "as is", meaning with no conversion. Currently, the data are read as an Uint8Array if the type is UNSIGNED_BYTE and as a Float32Array in all other cases. When using the new parameter, the buffer will be created as (U)Int8Array / (U)Int16Array / (U)Int32Array / Float32Array depending on the type of the texture.

### CODE REVIEW

File: src/Engines/thinEngine.ts [hunk 1]
```diff
@@ -4700,6 +4700,83 @@ export class ThinEngine {
         return internalFormat;
     }
 
+    /**
```
Comment (sebavan) [inline src/Engines/thinEngine.ts:4703]: Could this be moved from thinEngine to prevent growing it?

**Review by sebavan — CHANGES_REQUESTED** [review]

File: src/Engines/thinEngine.ts (hunk 1 above)
Comment (Popov72) [reply src/Engines/thinEngine.ts:4703]: I first put it in Misc/TextureTools.ts, but it means a dependency on this file from Extensions/engine.readTexture.ts: is that not a problem?

Comment (sebavan) [reply src/Engines/thinEngine.ts:4703]: so why not placing it in Extensions/engine.readTexture.ts ?

Comment (sebavan) [reply src/Engines/thinEngine.ts:4703]: but texture tools would be ok

Comment (Popov72) [reply src/Engines/thinEngine.ts]: Moved to TextureTools.

**Review by sebavan — APPROVED** [review]

File: src/Engines/thinEngine.ts (hunk 1 above)
Comment (Popov72) [reply src/Engines/thinEngine.ts]: Ok, it does not work to put it in `TextureTools`, so I have moved the code to `engine.readTexture`.

**Comment by sebavan** [comment]
One doc check to go and I ll merge @Popov72

**Pull request merged**